transform/mattermost-analytics/models/intermediate/product/nps/int_user_nps_score_latest.sql

```diff
@@ -0,0 +1,10 @@
+    SELECT
+        event_date as event_date,
+        server_id as server_id,
+        user_id as user_id,
+        server_version as server_version,
+        score as score,
+        user_role as user_role
+    FROM
+        {{ ref('int_nps_score') }} nps_score
+    qualify row_number() over (partition by server_id, user_id, server_version order by event_date desc) = 1
```

Comment:
This statement will keep the last nps score for each(user, server, server version). This is different from keeping the most recent comment.

Reply:
If they submitted more than 1 score for the same server_version, that means they went back into an old survey and submitted and updated score, this means a new survey wasn't sent since there was no version change and we update the model to the new score submitted. 

If there is a version update, a new survey will be sent and the old versions last score, and the new versions score, both will be preserved, the new score will be the recent score submitted.

Reply:
IMO, we shouldn't have more than 1 score for any given server_version, we would be tracking more than 1 score for a version if we did that.

Reply:
The new survey will be sent 21 days after the version upgrade, and only if the user hasn't completed a survey in the past 90 days. This allows for a window of the user modifying their feedback after the version upgrade.